File: xgboost/final.py
#! /usr/bin/env python

from format_data import formatData
from train_xgboost import train
import csv

def loadFinalData(flag):
	dataCSV = csv.reader(open("../data/data_"+flag+".csv", "r"))
	data = [row for row in dataCSV]
	formatData(data[1:], flag)

loadFinalData("train")
loadFinalData("test")
train("final")

File: final.R
#! /usr/bin/env Rscript

question <- commandArgs(TRUE)[1]
model <- commandArgs(TRUE)[2]

source("rfsrc/factor.R")
source(paste(model, "/train_", question, ".R", sep=""))

dtrain <- read.csv("data/data_train.csv")
dtest <- read.csv("data/data_test.csv")
if (model=="rfsrc")
{
	dtrain <- toFactors(dtrain)
	dtest <- toFactors(dtest)
}

dpred <- train(dtrain, dtest)
if (question=="1")
{
	write.table(round(dpred$a, 3), paste(model, "pred_1a_final.txt", sep="/"), quote=FALSE, row.names=FALSE, col.names=FALSE)
	write.table(round(dpred$b), paste(model, "pred_1b_final.txt", sep="/"), quote=FALSE, row.names=FALSE, col.names=FALSE)
}else
{
	write.table(round(dpred, 3), paste(model, "pred_2_final.txt", sep="/"), quote=FALSE, row.names=FALSE, col.names=FALSE)
}

